API/Repositories/Implementations/RepositoryBase.cs

```diff
@@ -13,9 +13,11 @@ protected RepositoryBase(OtrContext context)
         _context = context;
     }
 
-    public virtual async Task<T?> CreateAsync(T entity)
+    public virtual async Task<T> CreateAsync(T entity)
     {
-        var created = (await _context.Set<T>().AddAsync(entity)).Entity;
+        T? created = (await _context.Set<T>().AddAsync(entity)).Entity;
+        if (created == null)
+            throw new Exception($"Failed to create {nameof(T)} entity");
```

Comment:
hmm kinda annoyed the styler is letting this through. I prefer distinct braces for every control flow change.